src/_pytest/main.py

```diff
@@ -193,7 +193,11 @@ def wrap_session(
         except UsageError:
             session.exitstatus = ExitCode.USAGE_ERROR
             raise
-        except Failed:
+        except Failed as exc:
+            if not session.shouldfail:
+                # Typically Failed is raised when session.shouldfail has been
+                # set.  This sets it for when the exception was raised only.
+                session.shouldfail = exc.args[0]
```

Comment:
I usually use `str(exc)` instead of `exc.args[0]`, is that just style or is there a reason to prefer this way? (Just out of curiosity, it doesn't matter here of course).